lib/concurrent_guest_installations.pm

```diff
@@ -216,13 +214,7 @@ sub junit_log_provision {
         $_guest_installations_results->{$_}{stop_run} = ($guest_instances{$_}->{stop_run} eq '' ? time() : $guest_instances{$_}->{stop_run});
         $_guest_installations_results->{$_}{test_time} = strftime("\%Hh\%Mm\%Ss", gmtime($_guest_installations_results->{$_}{stop_run} - $_guest_installations_results->{$_}{start_run}));
     }
-    if (!version_utils::is_alp) {
-        $self->{"product_tested_on"} = script_output("cat /etc/issue | grep -io -e \"SUSE.*\$(arch))\" -e \"openSUSE.*[0-9]\"");
-    } else {
```

Comment:
Although currently sle micro 6 VT is rpm based, but it is hard to say for future SPs. Is it considerate to remove alp branches?

Reply:
Current alp code does not match the whole design and relevant code does not even tidy. I would like to clean them up. So they do not belong here. I do not want to keep useless and not tidy code. If future development needs it, someone can have a better look and implement them better and adapt them better.

Reply:
> Current alp code does not match the whole design

I think that's mainly because it is not quite been considered to be supported, and  instead been dropped, which I do not quite suggest.

Although the alp has been stopped in external communication, more and more further clarification seems that   it is NOT fully decided how to treat the alp code, used in sle16/sle micro or not. Our squad has been waiting for more technical decisions been clarified before deciding how to handle alp code in our automation -- reused partially, or refined, or deleted, or whatever proper based on real situation afterwards.  And when we decide to do it, we will have separate progress ticket for it. But, in this PR, it seems "already decided by you" to be fully deleted or break the code support. THIS WAY OF DOING TASK IS NOT SUGGESTED, AND THE TOUCHED SCOPE IS BEYOND THE TICKET SCOPE OF [poo#138635](https://progress.opensuse.org/issues/138635).

Reply:
I keep the alp lib file. But current alp code in these modules have to be removed, because:

    They are outdated already. For example, it still refers to "openSUSE" when generates junit.
    They are not clean and not conform to usage of parameters in base base. So they are not useful anymore.

The alp lib file is you largest asset which is kept. But the alp code must be cleaned up and added back (if necessary) in better form and format.


By the way, guest autoyast file already uses template, but it is not very suitable for guest profile.

Reply:
As ALP has not been finally decided to be stopped, I agree with keeping its support, or partial support, in our test. We need not create a totally new test from scratch since there were existing codes.

Reply:
This is not the case:
* Current alp code is already outdated and does not play a role in installing guest, and at the same time, it does not conform to the overall structure, so it has to be cleaned up.
* I ever mentioned that ```If there is new requirement, corresponding functionality will be added back with better form, nothing will be forgotten.``` which does not mean existing code can even partially support to some extent, what I meant is new requirement can be implemented in better way instead of inheriting "outdated and not good example" which distracts people from more proper way. So clean-up has to be done. 
* Please refer to [this comment](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/18488#discussion_r1502315656) again.
* Actually, all kinds of mediums are supported for guest installation, including pre-built image (raw/qcow/encrypted) and HTTP repo. There is no missing functionality (by contrast add support for pre-built image) compared with before doing clean-up. Clean-up here is to reorganizing/removing/improving code without affecting its original functionality. So you do not need to re-invent the same thing from scratch all over again. I did not remove useful functionality from existing code.